opm/input/eclipse/Generator/KeywordGenerator.cpp

```diff
@@ -17,83 +17,98 @@
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <fstream>
-#include <filesystem>
-#include <iostream>
-#include <sstream>
-#include <cctype>
-#include <fmt/format.h>
-
-#include <opm/json/JsonObject.hpp>
 #include <opm/input/eclipse/Generator/KeywordGenerator.hpp>
+
 #include <opm/input/eclipse/Generator/KeywordLoader.hpp>
+
+#include <opm/json/JsonObject.hpp>
+
 #include <opm/input/eclipse/Parser/ParserKeyword.hpp>
 
+#include <cctype>
+#include <filesystem>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <map>
+
+#include <fmt/format.h>
 
 namespace {
 
-const std::string sourceHeader = R"(
+    void updateFile(const std::string& newContent, const std::string& filename)
+    {
+        Opm::KeywordGenerator::ensurePath(filename);
+
+        std::ofstream { filename } << newContent;
+    }
+
+    void write_file(const std::string& content,
+                    const std::string& file,
+                    const bool         verbose,
+                    const std::string& desc = "source")
+    {
+        updateFile(content, file);
+
+        if (verbose) {
+            fmt::print("Updated {} file written to {}\n", desc, file);
+        }
+    }
+
+    void write_file(const std::stringstream& stream,
+                    const std::string&       file,
+                    const bool               verbose,
+                    const std::string&       desc = "source")
+    {
+        write_file(stream.str(), file, verbose, desc);
+    }
+
+    const std::string sourceHeader = R"(
 #include <opm/input/eclipse/Deck/UDAValue.hpp>
 #include <opm/input/eclipse/Parser/ParserItem.hpp>
 #include <opm/input/eclipse/Parser/ParserRecord.hpp>
 #include <opm/input/eclipse/Parser/Parser.hpp>
 
 
 )";
-}
+
+} // Anonymous namespace
 
 namespace Opm {
 
     KeywordGenerator::KeywordGenerator(bool verbose)
-        : m_verbose( verbose )
-    {
-    }
+        : m_verbose(verbose)
+    {}
 
-    std::string KeywordGenerator::headerHeader(const std::string& suffix) {
-        std::string header = fmt::format("#ifndef PARSER_KEYWORDS_{0}_HPP\n"
-                                         "#define PARSER_KEYWORDS_{0}_HPP\n"
-                                         "#include <opm/input/eclipse/Parser/ParserKeyword.hpp>\n"
-                                         "namespace Opm {{\n"
-                                         "namespace ParserKeywords {{\n\n", suffix);
-        return header;
+    std::string KeywordGenerator::headerHeader(const std::string& suffix)
+    {
+        return fmt::format("#ifndef PARSER_KEYWORDS_{0}_HPP\n"
+                           "#define PARSER_KEYWORDS_{0}_HPP\n"
+                           "#include <opm/input/eclipse/Parser/ParserKeyword.hpp>\n"
+                           "namespace Opm {{\n"
```

Comment:
may as well modernize, ie ```namespace Opm::ParserKeywords```

Reply:
> may as well modernize, i.e., `namespace Opm::ParserKeywords`

Agreed.  I've pushed an update to that effect.